From: YourName <you@example.com>
Date: Sun, 17 Aug 2025 23:59:00 +0200
Subject: [PATCH] soong: Add Android 13 (API 33) sepolicy compatibility

--- a/build/soong/android/config.go
+++ b/build/soong/android/config.go
@@ -1338,6 +1338,11 @@ func (c *config) VendorApiLevel() string {
        vendorApiLevel, err := strconv.Atoi(c.VendorApiLevelRaw())
        if err != nil {
                panic(fmt.Errorf("Cannot parse vendor API level %s to an integer: %s",
                        c.VendorApiLevelRaw(), err))
        }
+
+       if vendorApiLevel == 33 {
+               return "202404"
+       }
+
        return c.VendorApiLevelRaw()
 }
 
@@ -1348,6 +1353,11 @@ func (c *config) PrevVendorApiLevel() string {
                panic(fmt.Errorf("Cannot parse vendor API level %s to an integer: %s",
                        c.VendorApiLevel(), err))
        }
+
+       if c.VendorApiLevel() == "202404" {
+               return "33"
+       }
+
        if vendorApiLevel < 202404 || vendorApiLevel%100 != 4 {
                panic("Unknown vendor API level " + c.VendorApiLevel())
        }